trestle/core/control_reader.py

```diff
@@ -397,34 +397,42 @@ def _add_node_to_dict(
             # if it is a header, make sure it has correct format
             if node.key and node.key[0] == '#' and ControlInterface.bad_header(node.key):
                 raise TrestleError(f'Improper header format for control {control_id}: {node.key}')
-            prefix = '## ' if component_mode else '### '
-            if node.key.startswith(prefix):
-                if len(node.key.split()) <= 1:
-                    raise TrestleError(
-                        f'Header line in control {control_id} markdown starts with {prefix} but has no content.'
-                    )
-                comp_name = node.key.split(' ', 1)[1].strip()
-                simp_comp_name = ControlReader.simplify_name(comp_name)
-                if simp_comp_name == ControlReader.simplify_name(const.SSP_MAIN_COMP_NAME) and not component_mode:
-                    raise TrestleError(
-                        f'Response in control {control_id} has {const.SSP_MAIN_COMP_NAME} as a component heading.  '
-                        'Instead, place all response prose for the default component at the top of the section, with '
-                        'no ### component_name heading.  It will be entered as prose for the default system component.'
-                    )
-                if simp_comp_name in comp_list:
-                    raise TrestleError(
-                        f'Control {control_id} has a section with two component headings for {comp_name}.  '
-                        'Please combine the sections so there is only one heading for each component in a statement.'
-                    )
-                comp_list.append(simp_comp_name)
-                comp_name = ControlReader._comp_name_in_dict(comp_name, comp_dict)
+            if not component_mode:
+                # look for component name heading if present
+                prefix = '### '
+                if node.key.startswith(prefix):
```

Comment:
Simple test case here, user deletes the space after ###, this fails?

Reply:
This situation is caught by the bad_header function that requires a space after the last #.  I have added a new test file and test to cover that case.